nixd/tools/nixd/test/hover-options.md

```diff
@@ -0,0 +1,83 @@
+# RUN: nixd --lit-test \
+# RUN: --nixos-options-expr="{ foo.bar = { _type = \"option\"; }; }" \
+# RUN: < %s | FileCheck %s
+
+
+<-- initialize(0)
+
+```json
+{
+   "jsonrpc":"2.0",
+   "id":0,
+   "method":"initialize",
+   "params":{
+      "processId":123,
+      "rootPath":"",
+      "capabilities":{
+      },
+      "trace":"off"
+   }
+}
+```
+
+
+<-- textDocument/didOpen
+
+```json
+{
+   "jsonrpc":"2.0",
+   "method":"textDocument/didOpen",
+   "params":{
+      "textDocument":{
+         "uri":"file:///basic.nix",
+         "languageId":"nix",
+         "version":1,
+         "text":"{ foo.bar = 1 }"
+      }
+   }
+}
+```
+
+<-- textDocument/hover(2)
+
+
+```json
+{
+   "jsonrpc":"2.0",
+   "id":2,
+   "method":"textDocument/hover",
+   "params":{
+      "textDocument":{
+         "uri":"file:///basic.nix"
+      },
+      "position":{
+         "line":0,
+         "character":6
+      }
+   }
+}
+```
+
+```
+     CHECK: "id": 2,
+CHECK-NEXT: "jsonrpc": "2.0",
+CHECK-NEXT: "result": {
+CHECK-NEXT:   "contents": {
+CHECK-NEXT:     "kind": "markdown",
+CHECK-NEXT:     "value": "`Identifer`"
```

Comment:
No it didn't. I apologise but I couldn't figure out how to actually test a docstring.

Reply:
They are not actually "docstring"s but real regression tests, it will be checked in the CI. It is "markdown" because I think the content is more readable on github, with code blocks syntax-highlighted.

See https://github.com/nix-community/nixd/pull/526#pullrequestreview-2128670655 for how to update the test.

Reply:
I'm confused on how to proceed. Using this in the test gets `` `Identifer` `` as the value but I don't know how to change it or access nixos configs in the tests.

```json
{
   "jsonrpc":"2.0",
   "method":"textDocument/didOpen",
   "params":{
      "textDocument":{
         "uri":"file:///basic.nix",
         "languageId":"nix",
         "version":1,
         "text":"What do I put here?"
      }
   }
}
```

Reply:
> . Using this in the test gets `Identifer` as the value 

Make sure the nixd version is the binary just built in your development. Not system-wide binary. It should produce expected outputs.

> change it or access nixos configs in the tests.

Don't access any nixos configuration in unit tests. Just use `--nixos-options-expr` to mock an option attribute set.